hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/protocol/datatransfer/sasl/CustomizedCallbackHandler.java

```diff
@@ -34,6 +36,24 @@ public void handleCallback(List<Callback> callbacks, String username, char[] pas
     }
   }
 
+  static CustomizedCallbackHandler delegate(Object delegated) {
```

Comment:
As of last week, DynMethod exists in hadoop common to assist here. Have look to see if it would help or are your needs better. A key aspect is can extract IOEs from the invocation